h2o-algos/src/main/java/hex/pca/PCA.java

```diff
@@ -314,18 +309,23 @@ public void computeImpl() {
           // Compute SVD of Gram A'A/n using JAMA library
           // Note: Singular values ordered in weakly descending order by algorithm
           _job.update(1, "Calculating SVD of Gram matrix locally");
-          Matrix gramJ = _wideDataset ? new Matrix(ogtsk._gram.getXX()) : new Matrix(gtsk._gram.getXX());
-          SingularValueDecomposition svdJ = gramJ.svd();
+          DenseMatrix gramJ = _wideDataset ? new DenseMatrix(ogtsk._gram.getXX()) : new DenseMatrix(gtsk._gram.getXX());
+          int gramDimension = gramJ.numRows();
+          no.uib.cipr.matrix.SVD svdJ;
+          try {
+            // Note: gramJ will be overwritten after this
+            svdJ = new no.uib.cipr.matrix.SVD(gramDimension, gramDimension).factor(gramJ);
```

Comment:
@mmalohlava 
Should I also create a JIRA for this modularization of SVD method? It'd be useful for comparing various methods on-fly (dense vs. sparse matrix, Jama vs. netlib-java).